tika-pipes-core/src/main/java/org/apache/tika/pipes/core/parser/ParseService.java

```diff
@@ -84,12 +85,17 @@ public List<Map<String, Object>> parseDocument(InputStream inputStream, ParseCon
     }
 
     private RecursiveParserWrapperHandler getRecursiveParserWrapperHandler(ParseContext context, TikaConfig tikaConfig) {
-        HandlerConfig handlerConfig = new HandlerConfig(BasicContentHandlerFactory.HANDLER_TYPE.TEXT, HandlerConfig.PARSE_MODE.RMETA, writeLimit, maxEmbeddedResources, throwOnWriteLimitReached);
+        HandlerConfig handlerConfig = new HandlerConfig(getHandlerType(context), HandlerConfig.PARSE_MODE.RMETA, writeLimit, maxEmbeddedResources, throwOnWriteLimitReached);
         BasicContentHandlerFactory.HANDLER_TYPE type = handlerConfig.getType();
         BasicContentHandlerFactory contentHandlerFactory = new BasicContentHandlerFactory(type, handlerConfig.getWriteLimit(), handlerConfig.isThrowOnWriteLimitReached(), context);
         return new RecursiveParserWrapperHandler(contentHandlerFactory, handlerConfig.getMaxEmbeddedResources(), tikaConfig.getMetadataFilter());
     }
 
+    private BasicContentHandlerFactory.HANDLER_TYPE getHandlerType(ParseContext parseContext) {
```

Comment:
Nice how simple the solution is, although I am sure it required a bit of digging! (not discounting the effort at all)